instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -218,11 +218,6 @@ def _before_cur_exec(
             kind=trace.SpanKind.CLIENT,
         )
         with trace.use_span(span, end_on_exit=False):
-            if span.is_recording():
-                span.set_attribute(SpanAttributes.DB_STATEMENT, statement)
-                span.set_attribute(SpanAttributes.DB_SYSTEM, self.vendor)
-                for key, value in attrs.items():
-                    span.set_attribute(key, value)
             if self.enable_commenter:
```

Comment:
Does this code had any effect before the change? Like the changed statement is then reused later?

Reply:
I don't think it's reused. This method `_before_cur_exec` (that calls [_add_sql_comment](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/07c3324a3bcf80ee716f90b2456f59af08b685fa/opentelemetry-instrumentation/src/opentelemetry/instrumentation/sqlcommenter_utils.py#L19-L30) below on new L243) is only registered to SQLAlchemy [before_cursor_execute](https://docs.sqlalchemy.org/en/20/core/events.html#sqlalchemy.events.ConnectionEvents.before_cursor_execute) via SQLAlchemy [native event registration](https://docs.sqlalchemy.org/en/20/core/event.html#event-registration). The executed statement that includes sqlcomment gets logged by db server (if configured). Afaik, the query result returned to client does not include the sqlcomment (or any comment) so it doesn't get passed back that way.

Reply:
If it's not reused maybe we can move its calculation under `span.is_recording`?

Reply:
Yes! Updated in [e3ddea9](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2937/commits/e3ddea996242d2466e56cd78a061b945f810b288). Still works to update `db.statement` and add sql comment.